File: BACKEND/src/main/java/com/ROOMIFY/Roomify/model/Furniture.java
package com.ROOMIFY.Roomify.model;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "furniture")
@Getter
@Setter
public class Furniture {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String title;
    
    @Column(columnDefinition = "TEXT")
    private String description;
    
    private double price;
    private String category;
    private String conditionStatus; // NEW, USED
    
    private Long postedBy;
    private String ownerName;
    private String contactPhone;
    private String contactEmail;
    
    private double latitude;
    private double longitude;
    private String address;
    
    @ElementCollection(fetch = FetchType.EAGER)
    @CollectionTable(name = "furniture_images", joinColumns = @JoinColumn(name = "furniture_id"))
    @Column(name = "image")
    private List<String> images = new ArrayList<>();
    
    private String status = "AVAILABLE"; // AVAILABLE, SOLD
    
    private LocalDateTime createdAt = LocalDateTime.now();

    public Furniture() {}
}

File: FRONTEND/shared/src/commonMain/kotlin/org/com/model/Furniture.kt
package org.com.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

import org.com.network.ApiClient

@Serializable
data class Furniture(
    @SerialName("id")
    val id: Long? = null,
    @SerialName("title")
    val title: String = "",
    @SerialName("description")
    val description: String = "",
    @SerialName("price")
    val price: Double = 0.0,
    @SerialName("category")
    val category: String = "",
    @SerialName("conditionStatus")
    val condition: String = "", // NEW, USED
    @SerialName("postedBy")
    val postedBy: Long? = null,
    @SerialName("ownerName")
    val ownerName: String? = null,
    @SerialName("contactPhone")
    val contactPhone: String? = null,
    @SerialName("contactEmail")
    val contactEmail: String? = null,
    @SerialName("latitude")
    val latitude: Double = 0.0,
    @SerialName("longitude")
    val longitude: Double = 0.0,
    @SerialName("address")
    val address: String? = null,
    @SerialName("images")
    val images: List<String> = emptyList(),
    @SerialName("shopId")
    val shopId: Long? = null,
    @SerialName("videoUrl")
    val videoUrl: String? = null,
    @SerialName("hasVideo")
    val hasVideo: Boolean = false,
    @SerialName("createdAt")
    val createdAt: String? = null,
    @SerialName("status")
    val status: String = "AVAILABLE" // AVAILABLE, SOLD
) {
    val formattedPrice: String
        get() = "TZS ${price.toLong().toString().reversed().chunked(3).joinToString(",").reversed()}"

    val firstImageUrl: String?
        get() = images.firstOrNull()?.let { getFullUrl(it) }

    val fullImageUrls: List<String>
        get() = images.map { getFullUrl(it) }

    fun getFullUrl(path: String): String = ApiClient.resolveUrl(path)
}
